src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -380,7 +382,8 @@ export class TerminalService implements ITerminalService {
 		if (!offProcService) {
 			return this._availableProfiles || [];
 		}
-		return offProcService?.getProfiles(includeDetectedProfiles);
+		const platform = await this._getPlatformKey();
+		return offProcService?.getProfiles(this._workspaceTrustManagementService.isWorkpaceTrusted(), this._configurationService.getValue(`${TerminalSettingPrefix.Profiles}${platform}`), this._configurationService.getValue(`${TerminalSettingPrefix.DefaultProfile}${platform}`), includeDetectedProfiles);
```

Comment:
I don't think isWorkspaceTrusted is needed because getValue here will return the correct value?

Reply:
correct